app/scripts/controllers/taskDetails.js

```diff
@@ -57,4 +57,46 @@ angular.module('openhimConsoleApp')
     /**         Task Calculation Functions           **/
     /**************************************************/
 
-  });
\ No newline at end of file
+    function updateTaskWithStatus(task, status) {
+      var updated = new Api.Tasks();
+      updated._id = task._id;
```

Comment:
Maybe here you can just edit the task object and call `.$update` on it? Is there a need to make a new one?